src/views/Onboarding/OnboardingSetup.vue

```diff
@@ -44,7 +44,7 @@
       v-show="currentStep === 'password'"
       :imported="imported"
       @on-unlock="onUnlock"
-      @currentStep="currentStep = 'confirm'"
+      @currentStep="currentStep = imported ? $router.go(-1) : 'confirm'"
```

Comment:
something about this doesn't seem right. I can't understand what its doing because its setting and navigating at the same time
You should make this function separate and explicitly define the steps of behaviour such as 
```
if (currentStep === 'imported') router.go('/password')
else currentStep = 'confirm'
```

also try to avoid the $router.go(-1), usually its not very reliable. Route to a specific page you are looking to direct to

Reply:
All fixed.